pkg/tekton/metapipeline/clientfactory.go

```diff
@@ -0,0 +1,295 @@
+package metapipeline
+
+import (
+	"fmt"
+	"github.com/jenkins-x/jx/pkg/apps"
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	jxclient "github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/jxfactory"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/prow"
+	"github.com/jenkins-x/jx/pkg/tekton"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"os"
+	"time"
+)
+
+const (
+	defaultContainerImage = "jx"
+	defaultServiceAccount = "tekton-bot"
+	retryDuration         = time.Second * 30
+	defaultCheckoutDir    = "source"
+)
+
+var (
+	logger = log.Logger().WithFields(logrus.Fields{"component": "meta-pipeline-client"})
+)
+
+// ClientFactory implements the interface methods to create and apply the meta pipeline.
+type ClientFactory struct {
```

Comment:
This name is wired to me. `ClientFactory` is an implementation of `Client` interface.  Does it really need to be public, and exposed form the package? We can just return the `Client` interface from the constructor and you can name this just `client`.

Reply:
It was related to the default image and service account. And yes, we should just return Client. Let me sort this out.

Reply:
It would be great if we do not export the struct from the package by just using `client` without capital letter. 

I believe this is easy achievable if we return the interface in the constructor as suggested bellow. Also it is a static type check which ensure at compile time that the `client` type really implements the `Client` interface.

Reply:
done